### PR TITLE
RTL support per paragraph

### PR DESCRIPTION
**Pull request by belchevgb**

### CODE REVIEW

File: switch-text-direction/switch-text-direction.provider.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        name: "Right-to-left",
+        tooltip: "Right-to-left"
+    }
+};
+const arrowKeycodes = new Set([37, 38, 39, 40]);
+
+require("!style-loader!css-loader!./switch-text-direction.provider.css");
+
+@Injectable()
+class SwitchTextDirectionProvider implements ToolBarItemsProvider {
```
Comment (Novkirishki) [inline switch-text-direction/switch-text-direction.provider.ts]: Use **EditorConfigProvider**, **ToolBarItemsProvider** is depricated

File: switch-text-direction/switch-text-direction.provider.ts [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+                    this.handleButtonStylesOnLTRButtonClicked(editorHost);
+                }
+            }
+        };
+
+        const SWITCH_RIGHT_TO_LEFT_TOOL: ToolBarItem = {
+            name: TOOLBAR_BUTTONS_DATA.RTL.name,
+            tooltip:  TOOLBAR_BUTTONS_DATA.RTL.tooltip,
+            ordinal: 7,
+            exec: () => {
```
Comment (jordanilchev) [inline switch-text-direction/switch-text-direction.provider.ts]: let discuss if two exec functions can be DRY-ed

File: switch-text-direction/switch-text-direction.provider.ts [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    /**
+     * Determines if the passed element has display: inline.
+     *
+     * @private
+     * @param {HTMLElement} element The element which will be checked.
+     * @returns True if the element is inline, otherwise false.
+     * @memberof SwitchTextDirectionProvider
+     */
+    private isInlineElement(element: HTMLElement) {
+        const style = window.getComputedStyle(element, "");
```
Comment (jordanilchev) [inline switch-text-direction/switch-text-direction.provider.ts]: is it possible to not use ```window``` global object?

File: switch-text-direction/switch-text-direction.provider.ts [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+     * This gives access to the Kendo UI Editor configuration object
+     * that is used to initialize the editor upon creation
+     * Kendo UI Editor configuration Overiview documentation -> https://docs.telerik.com/kendo-ui/controls/editors/editor/overview#configuration
+     *
+     * @param {*} configuration
+     * @returns The modified configuration.
+     * @memberof SwitchTextDirectionProvider
+     */
+    configureEditor(configuration: any): any {
+        configuration.pasteCleanup.span = false;
```
Comment (Novkirishki) [inline switch-text-direction/switch-text-direction.provider.ts]: remove this

File: switch-text-direction/switch-text-direction.provider.ts [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+            if (!parent.classList.length) {
+                parent = parent.parentElement;
+            }
+
+            if (parent.classList.contains(LTR_CLASS)) {
+                this.getToolbarButton(editorHost, LTR_BUTTON_SELECTOR).classList.add(SELECTED_CLASS);
+                this.getToolbarButton(editorHost, RTL_BUTTON_SELECTOR).classList.remove(SELECTED_CLASS);
+            } else if (parent.classList.contains(RTL_CLASS) && !parent.classList.contains(KENDO_EDITOR_CLASS)) {
+                this.getToolbarButton(editorHost, RTL_BUTTON_SELECTOR).classList.add(SELECTED_CLASS);
+                this.getToolbarButton(editorHost, LTR_BUTTON_SELECTOR).classList.remove(SELECTED_CLASS);
```
Comment (Novkirishki) [inline switch-text-direction/switch-text-direction.provider.ts]: same as the above

File: switch-text-direction/switch-text-direction.provider.ts [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+            } else {
+                this.getToolbarButton(editorHost, LTR_BUTTON_SELECTOR).classList.remove(SELECTED_CLASS);
+                this.getToolbarButton(editorHost, RTL_BUTTON_SELECTOR).classList.remove(SELECTED_CLASS);
```
Comment (Novkirishki) [inline switch-text-direction/switch-text-direction.provider.ts]: same

File: switch-text-direction/switch-text-direction.provider.ts [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+            }
+        };
+
+        editorElement.addEventListener("click", () => toggleToolbarButtonsSelectedClasses());
+        editorElement.addEventListener("keydown", (ev: KeyboardEvent) => {
+            if (arrowKeycodes.has(ev.keyCode)) {
+                toggleToolbarButtonsSelectedClasses();
+            } else if (!editorElement.textContent) {
+                this.getToolbarButton(editorHost, LTR_BUTTON_SELECTOR).classList.remove(SELECTED_CLASS);
+                this.getToolbarButton(editorHost, RTL_BUTTON_SELECTOR).classList.remove(SELECTED_CLASS);
```
Comment (Novkirishki) [inline switch-text-direction/switch-text-direction.provider.ts]: same

File: switch-text-direction/switch-text-direction.provider.ts (hunk 6 above)
Comment (belchevgb) [reply switch-text-direction/switch-text-direction.provider.ts]: It's not applicable here.

**Pull request merged**